tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/graph/graph_component.ng.html

```diff
@@ -21,12 +21,121 @@
   </div>
 
   <div class="graph-structure-container">
-    <!-- TODO(cais): Add div for self op, input ops and consumer ops. -->
-    <!-- Make the following div "ng-template #noOpFocuse" of the else branch of an ngIf -->
+    <div *ngIf="opInfo !== undefined && opInfo !== null; else noOpFocused">
+      <div *ngIf="inputOps.length > 0; else noInputs" class="inputs-container">
+        <div>
+          <div
+            *ngFor="let inputOpInfo of inputOps; let slot = index"
+            class="input-op-section"
+          >
+            <div class="input-slot-header">
+              Input slot {{slot}}:
```

Comment:
I see some repetition. Would it make sense to create ng-template that is parameterized to take in `opInfo`?

Reply:
Also, I think it makes more sense to create `op` component that encapsulates some of these.

Reply:
Good suggestion. I have factor out the `op` part into an individual angular component,
namely the `GraphOpComponent` in the newly created `graph_op_component.ts`.

This improves the modularity of the code. Thanks.